server/user.go

```diff
@@ -299,36 +318,48 @@ func (s *UserMgmtServer) getCreds(r *http.Request, requiresAdmin bool) (api.Cred
 		return api.Creds{}, api.ErrorUnauthorized
 	}
 
-	usr, err := s.um.Get(sub)
-	if err != nil {
-		if err == user.ErrorNotFound {
-			return api.Creds{}, api.ErrorUnauthorized
-		}
-		log.Errorf("userMgmtServer: GetCreds err: %q", err)
-		return api.Creds{}, err
-	}
-
-	i := 0
-	for _, clientID := range clientIDs {
-		// Make sure the client actually exists.
-		isAdmin, err := s.cm.IsDexAdmin(clientID)
+	var usr user.User
+	if s.allowClientCredsAuth && (len(clientIDs) == 1) && (sub == clientIDs[0]) {
+		isAdmin, err := s.cm.IsDexAdmin(clientIDs[0])
 		if err != nil {
-			log.Errorf("userMgmtServer: GetCreds err: failed to get client %v", err)
+			log.Errorf("userMgmtServer: GetCreds err: %q", err)
 			return api.Creds{}, err
 		}
-
-		// If the endpoint requires an admin client, filter out clients which are not admins.
 		if requiresAdmin && !isAdmin {
-			continue
+			return api.Creds{}, api.ErrorForbidden
+		}
+	} else {
```

Comment:
nit: This else statement isn't needed. Just return after your last check in the if statement above an dedent the rest of this code.